libbeat/common/mapval/results.go

```diff
@@ -19,24 +19,40 @@ package mapval
 
 import "fmt"
 
+type resultsCache struct {
+	valid  *bool
+	errors *[]error
+}
+
 // Results the results of executing a schema.
 // They are a flattened map (using dotted paths) of all the values []ValueResult representing the results
 // of the IsDefs.
-type Results map[string][]ValueResult
+type Results struct {
+	Fields map[string][]ValueResult
+	// Lazy cache to prevent repeat computation
+	cache *resultsCache
+}
+
+func MakeResults() Results {
```

Comment:
Normally we call such methods `NewResults`

Reply:
++